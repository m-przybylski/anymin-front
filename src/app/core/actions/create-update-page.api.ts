// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';

export enum SessionUpdateApiActionTypes {
  CreateUpdateNameAndAvatar = '[create update session] Name and avatar',
}

export class CreateUpdateNameAndAvatarAction implements Action {
  public readonly type = SessionUpdateApiActionTypes.CreateUpdateNameAndAvatar;

  constructor(public payload: { name: string; avatarToken: string }) {}
}

export type SessionUpdateApiActionUnion = CreateUpdateNameAndAvatarAction;
