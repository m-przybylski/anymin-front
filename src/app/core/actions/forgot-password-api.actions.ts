import { Action } from '@ngrx/store';

export enum ForgotPasswordApiActionsTypes {
  SetMsisdnToken = '[API forgotPassword] SetMsisdnToken',
  DeleteMsisdnToken = '[API forgotPassword] DeleteMsisdnToken',
}

export class SetMsisdnToken implements Action {
  public readonly type = ForgotPasswordApiActionsTypes.SetMsisdnToken;

  constructor(public payload: string) {}
}

export class UnsetMsisdnTokenAction implements Action {
  public readonly type = ForgotPasswordApiActionsTypes.DeleteMsisdnToken;
}

export type ForgotPasswordApiActionsUnion = SetMsisdnToken | UnsetMsisdnTokenAction;
