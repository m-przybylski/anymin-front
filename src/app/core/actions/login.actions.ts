import { Action } from '@ngrx/store';
import { LoginCredentials, GetSessionWithAccount } from '@anymind-ng/api';

export enum AuthActionTypes {
  Login = '[auth] Login',
  LoginFromModal = '[auth] Login from modal',
  LoginFromModalSuccess = '[auth] Login from modal success',
  LoginError = '[auth] Login error',
  LoginSuccess = '[auth] Login success',
  Logout = '[auth] Logout',
  LogoutError = '[auth] Logout error',
  LogoutSuccess = '[auth] Logout success',
  LoginRedirect = '[auth] Login redirect',
  DashboardRedirect = '[auth] Dashboard redirect',
  LogoutRemote = '[auth] Logout by remote event',
  FirstLoginAfterRegistration = '[auth] First login and redirect after registration',
  UpdateFirstTimeLoginStatus = '[auth] First login and redirect update status',
}

export class LoginAction implements Action {
  public readonly type = AuthActionTypes.Login;

  constructor(public payload: LoginCredentials) {}
}

export class LoginModalAction implements Action {
  public readonly type = AuthActionTypes.LoginFromModal;

  constructor(public payload: LoginCredentials) {}
}

export class LoginErrorAction implements Action {
  public readonly type = AuthActionTypes.LoginError;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}
export class LoginModalSuccessAction implements Action {
  public readonly type = AuthActionTypes.LoginFromModalSuccess;
}

export class LoginSuccessAction implements Action {
  public readonly type = AuthActionTypes.LoginSuccess;

  constructor(public payload: GetSessionWithAccount) {}
}

export class LogoutAction implements Action {
  public readonly type = AuthActionTypes.Logout;
}
export class LogoutErrorAction implements Action {
  public readonly type = AuthActionTypes.LogoutError;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}
export class LogoutSuccessAction implements Action {
  public readonly type = AuthActionTypes.LogoutSuccess;
}
export class LoginRedirectAction implements Action {
  public readonly type = AuthActionTypes.LoginRedirect;
}
export class DashboardRedirectAction implements Action {
  public readonly type = AuthActionTypes.DashboardRedirect;

  constructor(public payload: GetSessionWithAccount) {}
}
export class LogoutRemoteAction implements Action {
  public readonly type = AuthActionTypes.LogoutRemote;
}
export class FirstLoginAfterRegistrationAction implements Action {
  public readonly type = AuthActionTypes.FirstLoginAfterRegistration;
}
export class UpdateFirstTimeLoginStatusAction implements Action {
  public readonly type = AuthActionTypes.UpdateFirstTimeLoginStatus;
}

export type AuthActionsUnion =
  | LoginAction
  | LoginErrorAction
  | LoginSuccessAction
  | LoginRedirectAction
  | LogoutAction
  | LogoutErrorAction
  | LogoutSuccessAction
  | DashboardRedirectAction
  | FirstLoginAfterRegistrationAction
  | UpdateFirstTimeLoginStatusAction
  | LoginModalAction
  | LoginModalSuccessAction
  | LogoutRemoteAction;
