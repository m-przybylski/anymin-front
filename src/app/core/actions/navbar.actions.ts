// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { UserTypeEnum } from '@platform/core/reducers/navbar.reducer';

export enum NavbarActionTypes {
  SetUserType = '[navbar] set user type',
  UpdateSessionAndUserType = '[navbar] set user type and fetch session',
  ToggleUserMenuVisibility = '[navbar] toggle user menu visibility',
  ToggleHelpMenuVisibility = '[navbar] toggle help menu visibility',
}

export class SetUserType implements Action {
  public readonly type = NavbarActionTypes.SetUserType;

  constructor(public payload: UserTypeEnum) {}
}

export class UpdateUserTypeAndSession implements Action {
  public readonly type = NavbarActionTypes.UpdateSessionAndUserType;

  constructor(public payload: UserTypeEnum) {}
}

export class ToggleUserMenuVisibility implements Action {
  public readonly type = NavbarActionTypes.ToggleUserMenuVisibility;
}

export class ToggleHelpMenuVisibility implements Action {
  public readonly type = NavbarActionTypes.ToggleHelpMenuVisibility;
}

export type NavbarActionsUnion =
  | SetUserType
  | UpdateUserTypeAndSession
  | ToggleUserMenuVisibility
  | ToggleHelpMenuVisibility;
