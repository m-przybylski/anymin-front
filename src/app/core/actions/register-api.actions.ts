import { Action } from '@ngrx/store';
import { GetSessionWithAccount } from '@anymind-ng/api';

export enum RegisterApiActionsTypes {
  RegisterError = '[API register] RegisterError',
  RegisterSuccess = '[API register] RegisterSuccess',
}

export class RegisterErrorAction implements Action {
  public readonly type = RegisterApiActionsTypes.RegisterError;

  constructor(public payload: any) {}
}

export class RegisterSuccessAction implements Action {
  public readonly type = RegisterApiActionsTypes.RegisterSuccess;

  constructor(public payload: GetSessionWithAccount) {}
}

export type RegisterAPIActionsUnion = RegisterErrorAction | RegisterSuccessAction;
