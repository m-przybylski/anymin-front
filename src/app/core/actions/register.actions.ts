import { Action } from '@ngrx/store';
import { GetSessionWithAccount, PostAccount } from '@anymind-ng/api';

export enum RegisterActionsTypes {
  Register = '[register] Register',
  RegisterByModal = '[register] Register by modal',
  RedirectToDashboardInvitations = '[register] Redirect to invitation list',
  RedirectToDashboardActivities = '[register] Register to activities',
}

export class RegisterAction implements Action {
  public readonly type = RegisterActionsTypes.Register;

  constructor(public payload: PostAccount) {}
}
export class RegisterByModalAction implements Action {
  public readonly type = RegisterActionsTypes.RegisterByModal;

  constructor(public payload: PostAccount) {}
}
export class RegisterRedirectToDashboardsInvitationsAction implements Action {
  public readonly type = RegisterActionsTypes.RedirectToDashboardInvitations;

  constructor(public payload: GetSessionWithAccount) {}
}

export class RegisterRedirectToDashboardsActivitiesAction implements Action {
  public readonly type = RegisterActionsTypes.RedirectToDashboardActivities;
}

export type RegisterActionsUnion =
  | RegisterAction
  | RegisterByModalAction
  | RegisterRedirectToDashboardsInvitationsAction
  | RegisterRedirectToDashboardsActivitiesAction;
