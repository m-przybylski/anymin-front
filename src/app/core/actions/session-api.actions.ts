// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { Account, GetSessionWithAccount } from '@anymind-ng/api';

export enum SessionWithAccountApiActionTypes {
  UpdateAccount = '[API Session with Account] update Account',
  FetchSessionError = '[API session] Fetch error',
  FetchSessionSuccess = '[API session] Fetch success',
  VerifyAccountByEmail = '[API session] Verify Account by Email',
  VerifyAccountByPin = '[API session] Verify Account By PIN',
}

export class UpdateAccountInSession implements Action {
  public readonly type = SessionWithAccountApiActionTypes.UpdateAccount;

  constructor(public payload: Account) {}
}

export class FetchSessionErrorAction implements Action {
  public readonly type = SessionWithAccountApiActionTypes.FetchSessionError;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}
export class FetchSessionSuccessAction implements Action {
  public readonly type = SessionWithAccountApiActionTypes.FetchSessionSuccess;

  constructor(public payload: GetSessionWithAccount) {}
}

export class VerifyAccountByEmailAction implements Action {
  public readonly type = SessionWithAccountApiActionTypes.VerifyAccountByEmail;

  constructor(public payload: GetSessionWithAccount) {}
}

export class VerifyAccountByPinAction implements Action {
  public readonly type = SessionWithAccountApiActionTypes.VerifyAccountByPin;

  constructor(public payload: GetSessionWithAccount) {}
}

export type SessionAPIActionUnion =
  | UpdateAccountInSession
  | FetchSessionErrorAction
  | FetchSessionSuccessAction
  | VerifyAccountByPinAction
  | VerifyAccountByEmailAction;
