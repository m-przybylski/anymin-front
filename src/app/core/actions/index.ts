import * as AuthActions from './login.actions';
import * as SessionActions from './session.actions';
import * as NavbarActions from './navbar.actions';
import * as SessionUpdateApiActions from './create-update-page.api';
import * as SessionApiActions from './session-api.actions';
import * as RegisterActions from './register.actions';
import * as RegisterApiActions from './register-api.actions';
import * as ForgotPasswordActions from './forgot-password-api.actions';
import * as SetNewPasswordActions from './set-new-password.actions';
import * as VisibilityInitActions from './visibility/visibility-init.actions';
import * as VisibilityApiActions from './visibility/visibility-api.actions';
import * as VisibilityUiActions from './visibility/visibility-ui.actions';
import * as VisibilityWSActions from './visibility/visibility-ws.actions';
import * as InvitationsApiActions from './invitations/invitations-api-actions';
import * as InvitationsWsActions from './invitations/invitations-ws-actions';
import * as InvitationsActions from './invitations/invitations-actions';

// import * as ActivitiesActions from './activities.actions';
// import * as ActivitiesApiActions from './activities/activities-api.actions';
// import * as ActivitiesPageActions from './activities/activities-page.actions';
// import * as BalanceApiActions from './activities/balance-api.actions';
// import * as ActivitiesWsActions from './activities/activities-ws.actions';

export {
  AuthActions,
  SessionActions,
  NavbarActions,
  SessionUpdateApiActions,
  SessionApiActions,
  RegisterActions,
  RegisterApiActions,
  ForgotPasswordActions,
  SetNewPasswordActions,
};
export { InvitationsApiActions, InvitationsWsActions, InvitationsActions };
export { VisibilityInitActions, VisibilityApiActions, VisibilityUiActions, VisibilityWSActions };

// export { ActivitiesActions, ActivitiesApiActions, ActivitiesPageActions, BalanceApiActions, ActivitiesWsActions };
