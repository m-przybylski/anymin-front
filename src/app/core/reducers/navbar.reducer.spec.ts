import { IState, reducer, UserTypeEnum } from './navbar.reducer';
import { AuthActions, NavbarActions } from '../actions';

describe('navbar.reducer', () => {
  const initialState: IState = {
    isNavbarHelpMenuVisible: false,
    isNavbarUserMenuVisible: false,
  };

  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toEqual(initialState);
    });
  });

  describe('SetUserType', () => {
    it('should set user type if no beginner state', () => {
      const result = reducer(initialState, new NavbarActions.SetUserType(UserTypeEnum.EXPERT));
      expect(result).toMatchSnapshot();
    });
    it('should update user type when state exists', () => {
      const state: IState = {
        isNavbarHelpMenuVisible: false,
        isNavbarUserMenuVisible: false,
        userType: UserTypeEnum.COMPANY,
      };
      const result = reducer(state, new NavbarActions.SetUserType(UserTypeEnum.EXPERT));
      expect(result).toMatchSnapshot();
    });
  });
  describe('UpdateSessionAndUserType', () => {
    it('should set user type if no beginner state', () => {
      const result = reducer(initialState, new NavbarActions.UpdateUserTypeAndSession(UserTypeEnum.EXPERT));
      expect(result).toMatchSnapshot();
    });
    it('should update user type when state exists', () => {
      const state: IState = {
        isNavbarHelpMenuVisible: false,
        isNavbarUserMenuVisible: false,
        userType: UserTypeEnum.COMPANY,
      };
      const result = reducer(state, new NavbarActions.UpdateUserTypeAndSession(UserTypeEnum.EXPERT));
      expect(result).toMatchSnapshot();
    });
  });
  describe('ToggleUserMenuVisibility', () => {
    it('should toggle menu visibility', () => {
      const result = reducer(initialState, new NavbarActions.ToggleUserMenuVisibility());

      expect(result).toMatchSnapshot();
    });
  });
  describe('ToggleHelpMenuVisibility', () => {
    it('should toggle help visibility', () => {
      const result = reducer(initialState, new NavbarActions.ToggleHelpMenuVisibility());

      expect(result).toMatchSnapshot();
    });
  });

  describe('LogoutSuccess', () => {
    it('should remove userType once logout', () => {
      const state: any = {
        userType: UserTypeEnum.COMPANY,
      };
      const result = reducer(state, new AuthActions.LogoutSuccessAction());

      expect(result).toMatchSnapshot();
    });
  });

  describe('LogoutRemote', () => {
    it('should remove userType once logout', () => {
      const state: any = {
        userType: UserTypeEnum.COMPANY,
      };
      const result = reducer(state, new AuthActions.LogoutRemoteAction());

      expect(result).toMatchSnapshot();
    });
  });
});
