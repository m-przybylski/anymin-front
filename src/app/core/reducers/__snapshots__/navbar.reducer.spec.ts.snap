// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar.reducer LogoutRemote should remove userType once logout 1`] = `
Object {
  "userType": undefined,
}
`;

exports[`navbar.reducer LogoutSuccess should remove userType once logout 1`] = `
Object {
  "userType": undefined,
}
`;

exports[`navbar.reducer SetUserType should set user type if no beginner state 1`] = `
Object {
  "isNavbarHelpMenuVisible": false,
  "isNavbarUserMenuVisible": false,
  "userType": 1,
}
`;

exports[`navbar.reducer SetUserType should update user type when state exists 1`] = `
Object {
  "isNavbarHelpMenuVisible": false,
  "isNavbarUserMenuVisible": false,
  "userType": 1,
}
`;

exports[`navbar.reducer ToggleHelpMenuVisibility should toggle help visibility 1`] = `
Object {
  "isNavbarHelpMenuVisible": true,
  "isNavbarUserMenuVisible": false,
}
`;

exports[`navbar.reducer ToggleUserMenuVisibility should toggle menu visibility 1`] = `
Object {
  "isNavbarHelpMenuVisible": false,
  "isNavbarUserMenuVisible": true,
}
`;

exports[`navbar.reducer UpdateSessionAndUserType should set user type if no beginner state 1`] = `
Object {
  "isNavbarHelpMenuVisible": false,
  "isNavbarUserMenuVisible": false,
  "userType": 1,
}
`;

exports[`navbar.reducer UpdateSessionAndUserType should update user type when state exists 1`] = `
Object {
  "isNavbarHelpMenuVisible": false,
  "isNavbarUserMenuVisible": false,
  "userType": 1,
}
`;
